view/frontend/web/js/product/button.js

```diff
@@ -0,0 +1,52 @@
+define([
+    'jquery',
+    'underscore'
+],function ($, _) {
+
+    return function (config) {
+
+        $(config.buttonId).on('mouseover',function (e) {
+
+            var radioAndCheckbox = {};
+            var inputElements = [];
+
+            $.each($('.product-options-wrapper [aria-required="true"]'),function () {
+
+                if(typeof $(this).prop("name") === "undefined" || $(this).attr("type") === "hidden"){
+                    return;
+                }
+
+                if($(this).attr("type") === "radio" || $(this).attr("type") === "checkbox"){
+
+                    String($(this).attr("name")) in radioAndCheckbox ?
+                        radioAndCheckbox[$(this).attr("name")].push($(this).is(':checked'))
+                        : radioAndCheckbox[$(this).attr("name")] = [$(this).is(':checked')]
+                    ;
+
+                } else {
+
+                    inputElements.push(Boolean($(this).val()));
```

Comment:
could you break out the `Boolean($(this).val())` part into a variable named like, isInputElementSet or something to be more explicit about what exactly is being added to this array, it's not super obvious at first glance

Reply:
I wouldn't change this, it's neater this way, and as soon as someone looks at $.each statement, it's obvious what it refers to.